### ISSUE TITLE
bug: preview broken if `:lcd` in oil buffer while previewing

### ISSUE THREAD

**Issue by Bekaboo** [stevearc/oil.nvim#215]
### Did you check the docs and existing issues?

- [X] I have read the docs
- [X] I have searched the existing issues

### Neovim version (nvim -v)

v0.10.0-dev-1452+g363e029e7ae

### Operating system/version

Linux 6.1.61-1-lts

### Describe the bug

Minimal config (just use default setup):

```lua
local tmp = vim.env.TMPDIR
  or vim.env.TEMPDIR
  or vim.env.TMP
  or vim.env.TEMP
  or '/tmp'
local data = tmp .. '/' .. (vim.env.NVIM_APPNAME or 'nvim')
local packages_root = data .. '/site'
local cloned_root = packages_root .. '/pack/packages/start'
local cloned_path = cloned_root .. '/oil.nvim'
local url = 'https://github.com/stevearc/oil.nvim.git'

vim.fn.mkdir(cloned_root, 'p')
vim.opt.pp:prepend(packages_root)
vim.opt.rtp:prepend(packages_root)

if not vim.loop.fs_stat(cloned_path) then
  vim.fn.system({ 'git', 'clone', url, cloned_path })
end

require('oil').setup()
```

After executing `:lcd` in a oil buffer when previewing, preview does not work. Seems that oil is using relative path instead of absolute path to find the location of the file being previewed.

Screen recording:

https://github.com/stevearc/oil.nvim/assets/76579810/ccf85133-ddd8-4fce-ab37-5cf2f4163771

### Steps To Reproduce

1. `rm -r ~/.local/share/nvim-test && rm -r ~/.local/state/nvim-test && rm -r ~/.config/nvim-test && mkdir -p ~/.config/nvim-test` and then save the config above as `~/.config/nvim-test/init.lua`
2. Create a test directory containing several files  with different contents
3. `NVIM_APPNAME=nvim-test nvim <test_dir>`
4. Press `<C-p>` to preview, works fine
5. Keep preview on, `:lcd <another dir>`
6. Move the cursor to next file
7. Preview does not work for other files

### Expected Behavior

Should be able to preview other files after `cd`/`lcd`/`tcd`

### Directory structure

a/1.txt
a/2.txt
a/3.txt
a/4.txt
a/5.txt

### Repro

```Lua
See above
```

### Did you check the bug with a clean config?

- [X] I have confirmed that the bug reproduces with `nvim -u repro.lua` using the repro.lua file above.

**Comment by stevearc** [comment]
This is probably also fixed by #214. It's still pending (waiting for at least a couple people to try it out and report if there are any bugs). If you want, give that PR a try and see if it fixes this for you

**Comment by Bekaboo** [comment]
That PR indeed solves the issue, thanks!
